packages/headless/src/plans.ts

```diff
@@ -11,3 +11,66 @@ export function usePlans(): UseQueryResult<IPlan.MappedAttributes[], Error> {
     queryKey: ["find-all-plans"],
   });
 }
+
+export function useCreatePlan({
+  onSuccess,
+  onError,
+}: {
+  onSuccess?(): void;
+  onError?(error: Error): void;
+}) {
+  const atlas = useAtlas();
+  const createPlan = useCallback(
+    (payload: IPlan.CreateApiPayload) => atlas.plan.create(payload),
+    [atlas.plan]
+  );
+  return useMutation({
+    mutationFn: createPlan,
+    mutationKey: ["create-plan"],
+    onSuccess,
+    onError,
+  });
+}
+
+export function useUpdatePlan({
+  id,
+  onSuccess,
+  onError,
+}: {
+  id: number;
+  onSuccess?: Void;
+  onError?(error: Error): void;
+}) {
+  const atlas = useAtlas();
+  const updatePlan = useCallback(
+    (payload: IPlan.UpdateApiPayload) => atlas.plan.update(id, payload),
+    [atlas.plan]
+  );
+
+  return useMutation({
+    mutationFn: updatePlan,
+    mutationKey: ["update-plan", id],
+    onSuccess,
+    onError,
+  });
+}
+
+export function useDeletePlan({
+  id,
+  onSuccess,
+  onError,
+}: {
+  id: number;
+  onSuccess?: Void;
+  onError?(error: Error): Void;
+}) {
```

Comment:
Don't pass the id to the hook.

```suggestion
export function useDeletePlan({
  onSuccess,
  onError,
}: {
  onSuccess?: Void;
  onError?(error: Error): Void;
}) {
```